### PR TITLE
[FEATURE] Control volume of metadata output when running DataAssistant classes.

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Provide a flag that controls the amount of `Rule-Based Profiler` configuration information, returned as part of the `meta["citations"]` field of `ExpectationSuite`, provided by `DataAssistantResult` object.
* Refactor the implementation of controlling the amount of `DataAssistantResult` output depending on whether or not the experience is in a Jupyter notebook (for code elegance).
* Update existing unit tests.

### Additional Scope
* Repair `Query` metrics to use appropriate data types (and update type hints accordingly).
* Repair `SqlAlchemyExecutionEngine` to handle WHERE clause properly.

### Note
Another quick exercise of User-Acceptance Testing would be beneficial.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-761/GREAT-1041
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM

**Comment by alexsherstinsky** [comment]
The following tests are failing only in Azure, but do not even run locally:
```
FAILED tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py::test_expect_queried_column_value_frequency_to_meet_threshold_sqlite_multi_value[test_backends0]
FAILED tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py::test_expect_queried_column_value_frequency_to_meet_threshold_sqlite[test_backends0-batch_request1-True-0.6481340441736482-None-False]
FAILED tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py::test_expect_queried_column_value_frequency_to_meet_threshold_sqlite[test_backends0-batch_request2-False-0.4791666666666667-col("Age")<18-True]
FAILED tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py::test_expect_queried_column_value_frequency_to_meet_threshold_override_query_sqlite[test_backends0-batch_request2-SELECT {col}, CAST(COUNT({col}) AS float) / (SELECT COUNT(y) FROM wrong) FROM {active_batch} GROUP BY {col}-True-7.091666666666667-None-True]
FAILED tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py::test_expect_queried_column_value_frequency_to_meet_threshold_override_query_sqlite[test_backends0-batch_request3-SELECT {col}, CAST(COUNT({col}) AS float) / (SELECT COUNT({col}) FROM titanic) FROM {active_batch} GROUP BY {col}-False-0.2338156892612338-col("Age")<35-True]
FAILED tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py::test_expect_queried_column_value_frequency_to_meet_threshold_override_query_sqlite[test_backends0-batch_request4-SELECT {col}, CAST(COUNT({col}) AS float) / (SELECT COUNT({col}) FROM {active_batch}) / 2 FROM {active_batch} GROUP BY {col}-False-0.3240670220868241-None-False]
FAILED tests/expectations/core/test_expect_queried_table_row_count_to_be.py::test_expect_queried_column_value_frequency_to_meet_threshold_sqlite[test_backends0-batch_request1-False-100-1313-None]
FAILED tests/expectations/core/test_expect_queried_table_row_count_to_be.py::test_expect_queried_column_value_frequency_to_meet_threshold_sqlite[test_backends0-batch_request2-False-100-96-col("Age")<18]
```

If the query expectations are supposed to be in `contrib`, how come we are seeing the test for it in the `core` section?

**Comment by rdodev** [comment]
> The following tests are failing only in Azure, but do not even run locally:
> 
> ```
> FAILED tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py::test_expect_queried_column_value_frequency_to_meet_threshold_sqlite_multi_value[test_backends0]
> FAILED tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py::test_expect_queried_column_value_frequency_to_meet_threshold_sqlite[test_backends0-batch_request1-True-0.6481340441736482-None-False]
> FAILED tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py::test_expect_queried_column_value_frequency_to_meet_threshold_sqlite[test_backends0-batch_request2-False-0.4791666666666667-col("Age")<18-True]
> FAILED tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py::test_expect_queried_column_value_frequency_to_meet_threshold_override_query_sqlite[test_backends0-batch_request2-SELECT {col}, CAST(COUNT({col}) AS float) / (SELECT COUNT(y) FROM wrong) FROM {active_batch} GROUP BY {col}-True-7.091666666666667-None-True]
> FAILED tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py::test_expect_queried_column_value_frequency_to_meet_threshold_override_query_sqlite[test_backends0-batch_request3-SELECT {col}, CAST(COUNT({col}) AS float) / (SELECT COUNT({col}) FROM titanic) FROM {active_batch} GROUP BY {col}-False-0.2338156892612338-col("Age")<35-True]
> FAILED tests/expectations/core/test_expect_queried_column_value_frequency_to_meet_threshold.py::test_expect_queried_column_value_frequency_to_meet_threshold_override_query_sqlite[test_backends0-batch_request4-SELECT {col}, CAST(COUNT({col}) AS float) / (SELECT COUNT({col}) FROM {active_batch}) / 2 FROM {active_batch} GROUP BY {col}-False-0.3240670220868241-None-False]
> FAILED tests/expectations/core/test_expect_queried_table_row_count_to_be.py::test_expect_queried_column_value_frequency_to_meet_threshold_sqlite[test_backends0-batch_request1-False-100-1313-None]
> FAILED tests/expectations/core/test_expect_queried_table_row_count_to_be.py::test_expect_queried_column_value_frequency_to_meet_threshold_sqlite[test_backends0-batch_request2-False-100-96-col("Age")<18]
> ```
> 
> If the query expectations are supposed to be in `contrib`, how come we are seeing the test for it in the `core` section?

I'm seeing the exact same failures in my PR. I wonder what's causing it?

**Comment by Shinnnyshinshin** [comment]
> test_expect_queried_table_row_count_to_be

I was able to re-create the error on my end and may have a fix. The issue appears to be how the fixture (`titanic.csv`) is being imported into the test.

File: tests/conftest.py [hunk 1]
```diff
@@ -859,6 +858,19 @@ def titanic_v013_multi_datasource_multi_execution_engine_data_context_with_check
     return context
 
 
+@pytest.fixture
+def titanic_v013_multi_datasource_multi_execution_engine_data_context_with_checkpoints_v1_with_empty_store_stats_enabled(
+    sa,
+    spark_session,
```
Comment (Shinnnyshinshin) [inline tests/conftest.py:864]: These were necessary to pass the `query` expectations

**Pull request merged**